server/controllers/testsController.js

```diff
@@ -0,0 +1,193 @@
+const testsController = {};
+
+/*
+  Middleware to verify that we have the properties we are looking for in our input object. If it fails, return 400 error code.
+  verify the reqest body has all the relevant properties and save them to res.locals.header and res.locals.assertions
+
+  EXPECTED INPUT:
+    {
+      header: {
+                endpoint : '/',
+                method: 'POST'
+                req_body: {a:1} // Needed for POST and PATCH, optional for DELETE
+      },
+      assertions: [
+                    { content: '/text\/html/' },
+                    { status: 200 },
+                    { res_body: { a: 'b' } }
+                    ...
+      ]
+    }
+    EXPECTED OUTPUT: res.locals.header, res.locals.assertions
+*/
+testsController.verifyInput = (req, res, next) => {
+  try {
+    const { header, assertions } = req.body;
+
+    // Check for improper input
+    if (
+      typeof header.endpoint !== 'string' ||
+      typeof header.method !== 'string' ||
+      !Array.isArray(assertions)
+    ) {
+      throw new Error('Missing properties or wrong data types in request.');
+    }
+
+    // Check for correct request methods
+    header.method = header.method.toUpperCase();
+    if (
+      header.method !== 'GET' &&
```

Comment:
refactoring idea to keep code DRY.  Consider storing allowable values in a set and check that the set contains header.method